code/game/objects/items/devices/multitool.dm

```diff
@@ -94,19 +105,51 @@
 // Syndicate device disguised as a multitool; it will turn red when an AI camera is nearby.
 
 /obj/item/multitool/ai_detect
-	actions_types = list(/datum/action/item_action/toggle_multitool)
+	/// How close the AI is to us
 	var/detect_state = PROXIMITY_NONE
+	/// Range at which the closest AI makes the multitool glow red
 	var/rangealert = 8 //Glows red when inside
+	/// Range at which the closest AI makes the multitool glow yellow
 	var/rangewarning = 20 //Glows yellow when inside
-	var/hud_type = DATA_HUD_AI_DETECT
-	var/detecting = FALSE
+	/// Is our HUD on
+	var/hud_on = FALSE
+
+/obj/item/multitool/ai_detect/examine(mob/user)
+	. = ..()
+	if(!hud_on)
+		return
+	. += span_notice("You can right-click to scan for a nearby APC.")
+	switch(detect_state)
+		if(PROXIMITY_NONE)
+			. += span_green("No AI should be currently looking at you. Keep on your clandestine activities.")
+		if(PROXIMITY_NEAR)
+			. += span_warning("An AI is getting uncomfortably close. Maybe time to drop what youre doing.")
+		if(PROXIMITY_ON_SCREEN)
+			. += span_danger("An AI is (probably) looking at you. You should probably hide this.")
 
 /obj/item/multitool/ai_detect/Destroy()
-	STOP_PROCESSING(SSfastprocess, src)
+	if(hud_on && ismob(loc))
+		remove_hud(loc)
 	return ..()
 
-/obj/item/multitool/ai_detect/ui_action_click()
-	return
+/obj/item/multitool/ai_detect/attack_self(mob/user, modifiers)
+	apc_scanner = FALSE //we want to toggle hud, not check for APC
```

Comment:
the apc scanning function should be assigned to one of the two click procs but not both. I'd suggest setting the variable to FALSE on compile and have the action done in attack_self_secondary() instead.